app/helpers/data_summary_helper.rb

```diff
@@ -102,17 +108,31 @@ def append_key_value_to_hash(sorted_list)
   end
 
   def append_number_of_times_current_value_appears(input_value, input_hash)
+    # because other is always put the last so we only have to detect the first value of current index is 'Other:'
+    other_value_starts = false
+    other_text = 'Other:'
+
     # if it is not string do no nothing
     if input_value.instance_of?(String)
-      # get data from each field - list
-      values = input_value.split(';', -1)
-      # for loop each item in the list
-      values.each do |value|
+      # example of input_value -> A, B, C,Other: I am an ,other value,Other: I am another other value
+      # input_value.split(';', -1) -> ['A', 'B', 'C', 'Other: I am an', 'other value','Other: I am another other value']
+      input_value.split(';', -1).each do |value|
+        first_six_characters = value[0, 6]
         # If the value is :Other then aggregate it to Others
-        is_other = value[0, 6] == 'Other:'
-        current_key = is_other ? 'Others' : value.strip
-        input_hash[current_key] = 0 if input_hash[current_key].nil?
-        input_hash[current_key] += 1
+        other_value_starts = true if first_six_characters == other_text
```

Comment:
We can also check this way as well: value.start_with?(other_text) 
This will compare the starting letters. 

```suggestion
    value.strip!
    next if value.empty?

    if value.start_with?(other_text)
```